pipeline/gpg_signing/signing.go

```diff
@@ -0,0 +1,129 @@
+// Package checksums provides a Pipe that creates .checksums files for
+// each artifact.
+package gpg_signing
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/apex/log"
+	"github.com/mitchellh/go-homedir"
+	"golang.org/x/crypto/openpgp"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/goreleaser/goreleaser/context"
+	"github.com/goreleaser/goreleaser/pipeline"
+)
+
+// Pipe for checksums
+type Pipe struct{}
+
+// Description of the pipe
+func (Pipe) Description() string {
+	return "Signing release artifacts"
+}
+
+// Run the pipe
+func (Pipe) Run(ctx *context.Context) (err error) {
+	if ctx.Config.GPGSigning.KeyID == "" {
+		return pipeline.Skip("gpg_signing.key_id is not configured")
+	}
+
+	// read key id from config
+	if len(ctx.Config.GPGSigning.KeyID) != 16 {
+		return fmt.Errorf("invalid key_id '%s', needs to be a 8 byte long hex key id", ctx.Config.GPGSigning.KeyID)
+	}
+	keyID, err := strconv.ParseUint(ctx.Config.GPGSigning.KeyID, 16, 64)
+	if err != nil {
+		return fmt.Errorf("invalid key_id '%s': %s", ctx.Config.GPGSigning.KeyID, err)
+	}
+
+	// read gpg key ring
+	keyRingPath, err := homedir.Expand("~/.gnupg/secring.gpg")
+	if err != nil {
+		return err
+	}
+	keyRingFile, err := os.OpenFile(
+		keyRingPath,
+		os.O_RDONLY,
+		0644,
+	)
+	if err != nil {
+		return err
+	}
+	defer keyRingFile.Close()
```

Comment:
this could err...

Reply:
Yes, but what are you going to do with the error? The operation has already completed successfully. In general, I ignore `file.Close()` errors unless the code depends on the file being closed.

Reply:
I usually only `log.WithError(err).Errorf("failed to close file %s", file)`...